commands/scan/scan.go

```diff
@@ -204,7 +207,7 @@ func hasViolationContext(watches []string, projectKey string) bool {
 	return len(watches) > 0 || projectKey != ""
 }
 
-func (scanCmd *ScanCommand) RunAndRecordResults(recordResFunc func(scanResults *utils.Results) error) (err error) {
+func (scanCmd *ScanCommand) RunAndRecordResults(cmdType utils.CommandType, recordResFunc func(scanResults *utils.Results) error) (err error) {
```

Comment:
can this function be split a bit more?
It's over 100 lines - and looks like doing logic in multiple ares

Reply:
it will be changed after the refactor, not relevant for this PR